arbnode/transaction_streamer.go

```diff
@@ -773,6 +777,29 @@ func (s *TransactionStreamer) createBlocks(ctx context.Context) error {
 	return nil
 }
 
+func (s *TransactionStreamer) SyncProgressMap() map[string]interface{} {
+	batchSeen := s.inboxReader.GetLastSeenBatchCount()
+	_, batchProcessed := s.inboxReader.GetLastReadBlockAndBatchCount()
+	broadcasterQueuedMessagesPos := atomic.LoadUint64(&s.broadcasterQueuedMessagesPos)
+
+	res := make(map[string]interface{})
+
+	if batchProcessed >= batchSeen && (broadcasterQueuedMessagesPos == 0) && (batchSeen > 0) {
+		return res
```

Comment:
One other thing we should check here is if the latest block is caught up to the latest batch's last message (TxStreamer + BlockChain should have all the info you need for that)

Reply:
We might also want some sort of allowed gap where if you're caught up to the last seen state from 5 seconds ago you're seen as synced or something

Reply:
My theory:
- a node that's in-line should be fully updated within one internal loop of inbox_reader.
- lastSeen is only updated after lastprocessed (or will be,after I fix a small race condition :)), thus reading lastseen before lastprocessed should be enough. There shouldn't be any gap